ime/app/src/main/java/com/anysoftkeyboard/AnySoftKeyboard.java

```diff
@@ -86,6 +86,7 @@
 
     private boolean mAutoCap;
     private boolean mKeyboardAutoCap;
+    private boolean mShouldAutoSpaceForPunctuation;
```

Comment:
I believe it will make more sense to have this change done in the `AnySoftKeyboardSuggestions` class.

Reply:
I agree with you, that's why I started exactely from there but I got stuck. 

```
        if (TextEntryState.isPredicting() && !separatorInsideWord) {
            // ACTION does not invoke default picking. See
            // https://github.com/AnySoftKeyboard/AnySoftKeyboard/issues/198
            pickDefaultSuggestion(isAutoCorrect() && !newLine);
            // Picked the suggestion by a space/punctuation character: we will treat it
            // as "added an auto space".
            mAdditionalCharacterForReverting = !newLine;
```
Whatever modifications I make with `ic.deleteSurroundingText` and `ic.commitText` after `pickDefaultSuggestion` method call, will be applied before the `primarycode`. I didn't find any way  to add a space AFTER `primarycode` and not BEFORE. Whatever I do goes before!

Making changes in `AnySoftwareKeyboard` after `handleSeparator(primaryCode)` call magically works! :) I'm sure there is a way, please advice me :)

_If there in no other way_, I'd rather delete the `swapping punctuation and space` involved code and keep this new feature.

btw, I also extended the code to make it work with French punctuation if keyboard locale is fr ;) I'll push it after we decide how to apply the requested changes.

Reply:
I think I got where I was in error. I have a better understanding of the code now. I can try to fix it now ;)